### PR TITLE
Trace group

### PR DESCRIPTION
**Pull request by teetak01**
@jupe @kuggenhoffen @anttiylitokola 

Could you check this? This now allows specification of a TRACE_GROUP name from config.json. It also now has default TRACE_GROUP "APPL" in case TRACE_GROUP is not defined anywhere.

For. example

{
    "mbed-trace": 1,
    "mbed-trace-group": "mbed-client"
}

Also added the NDEBUG flag to checks. Otherwise compiling with '-d' flag does not enable traces.

The HAVE_DEBUG flag mentioned in mbed Client m2mconfig.h is obsolete and does not seem to do anything. That if statement is purely decided by NDEBUG flag.

### CODE REVIEW

File: source/mbed_trace.c [hunk 1]
```diff
@@ -259,6 +259,9 @@ static void mbed_trace_default_print(const char *str)
 }
 void mbed_tracef(uint8_t dlevel, const char *grp, const char *fmt, ...)
 {
+    if (NULL == m_trace.line) {
```
Comment (jupe) [inline source/mbed_trace.c]: this is not relatedto pr at all. should be separate pr.

Comment (teetak01) [inline source/mbed_trace.c]: Moved to another PR

File: mbed-trace/mbed_trace.h [hunk 2]
```diff
@@ -110,7 +110,21 @@ extern "C" {
...
+#if defined(YOTTA_CFG_MBED_TRACE) || (defined(YOTTA_CFG) && !defined(NDEBUG))
+
+/**
+ * Allow specification of default TRACE_GROUP to be used if not specified by application
+ */
+
+#ifndef TRACE_GROUP
+#ifdef YOTTA_CFG_MBED_TRACE_GROUP
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
```
Comment (jupe) [inline mbed-trace/mbed_trace.h]: this not look so "safe" way to do this I think. It's quite big possibility that somebody else (end-user) has define `STR` already, so this fall down because duplicate define. perhaps there is better solution for this

Comment (SeppoTakalo) [inline mbed-trace/mbed_trace.h]: Yes, I agree.
This is public header file, so you are polluting public namespace with macros `STR` and `STR_HELPER` which are too generic, I think.

Refactor those to be `TRACE_GROUP_STR` and `TRACE_GROUP_STR_HELPER`

**Comment by teetak01** [comment]
Renamed macros
